UE/Application/Ports/UserPort.cpp

```diff
@@ -160,10 +160,22 @@ void UserPort::setupIncomingCallButtons(std::function<void()> acceptButtonCallba
     gui.setRejectCallback(rejectButtonCallback);
 }
 
-void UserPort::showTalking()
+void UserPort::showTalking(std::string& text)
 {
     logger.logInfo("UserPort::showTalking");
-    gui.setCallMode();
+    auto& talking = gui.setCallMode();
+    if(text != ""){
+        talking.appendIncomingText(text);
+    }
+    gui.setAcceptCallback([&](){
+        logger.logInfo("UserPort::Talking ", talking.getOutgoingText());
+        std::string newText = to_string(phoneNumber) + ": " + talking.getOutgoingText();
+        handler->handleTalkTextSend(newText);
```

Comment:
R-value reference could be possible here as well  - but optional, since we have to create a new string anyways.